## 1. The World Bank's international debt data
Humans don't solely incur debt to meet basic needs; countries also take on debt to manage their economies effectively. For instance, infrastructure investment is a significant expense necessary for ensuring a high quality of life for citizens. The World Bank is a key institution that provides loans to nations.

In this notebook, we will analyze international debt data gathered by The World Bank. The dataset includes details about the debt (in USD) owed by developing countries across various categories. We aim to address questions such as:

<ul>
<li>What is the total amount of debt that is owed by the countries listed in the dataset?</li>
<li>Which country owns the maximum amount of debt and what does that amount look like?</li>
<li>What is the average amount of debt owed by countries across different debt indicators?</li>
</ul>


In [ ]:
%%sql
postgresql:///international_debt

SELECT * FROM international_debt
LIMIT 10;



10 rows affected.


[('Afghanistan', 'AFG', 'Disbursements on external debt, long-term (DIS, current US$)', 'DT.DIS.DLXF.CD', Decimal('72894453.700000003')),
 ('Afghanistan', 'AFG', 'Interest payments on external debt, long-term (INT, current US$)', 'DT.INT.DLXF.CD', Decimal('53239440.100000001')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (AMT, current US$)', 'DT.AMT.BLAT.CD', Decimal('61739336.899999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (DIS, current US$)', 'DT.DIS.BLAT.CD', Decimal('49114729.399999999')),
 ('Afghanistan', 'AFG', 'PPG, bilateral (INT, current US$)', 'DT.INT.BLAT.CD', Decimal('39903620.100000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (AMT, current US$)', 'DT.AMT.MLAT.CD', Decimal('39107845')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (DIS, current US$)', 'DT.DIS.MLAT.CD', Decimal('23779724.300000001')),
 ('Afghanistan', 'AFG', 'PPG, multilateral (INT, current US$)', 'DT.INT.MLAT.CD', Decimal('13335820')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (AMT, current US$)', 'DT.AMT.OFFT.CD', Decimal('100847181.900000006')),
 ('Afghanistan', 'AFG', 'PPG, official creditors (DIS, current US$)', 'DT.DIS.OFFT.CD', Decimal('72894453.700000003'))]

## 2. The number of distinct countries
Examining the first ten rows, we can observe the debt amounts owed by Afghanistan across various debt indicators. However, we lack information on the total number of distinct countries represented in the dataset. The repetition of country names suggests that a nation may have debt in multiple indicators.

To conduct comprehensive statistical analyses, it's essential to determine the count of unique countries in the dataset. In this section, we will extract the number of distinct countries present in the table.

In [ ]:
%%sql
SELECT COUNT(DISTINCT(country_name)) AS total_distinct_countries
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(124,)]

## 3. The distinct debt indicators
The dataset includes a total of 124 countries. As mentioned in the first section, there is a column titled <code>indicator_name</code> that provides a brief explanation of the reasons for incurring debt. Next to this column is another one called <code>indicator_code</code>, which denotes the category of these debts. Familiarity with these different debt indicators will aid in understanding the specific areas where a country may have outstanding debts.

In [ ]:
%%sql

SELECT DISTINCT(indicator_code) AS distinct_debt_indicators
FROM international_debt
ORDER BY 1;

 * postgresql:///international_debt
25 rows affected.


[('DT.AMT.BLAT.CD',),
 ('DT.AMT.DLXF.CD',),
 ('DT.AMT.DPNG.CD',),
 ('DT.AMT.MLAT.CD',),
 ('DT.AMT.OFFT.CD',),
 ('DT.AMT.PBND.CD',),
 ('DT.AMT.PCBK.CD',),
 ('DT.AMT.PROP.CD',),
 ('DT.AMT.PRVT.CD',),
 ('DT.DIS.BLAT.CD',),
 ('DT.DIS.DLXF.CD',),
 ('DT.DIS.MLAT.CD',),
 ('DT.DIS.OFFT.CD',),
 ('DT.DIS.PCBK.CD',),
 ('DT.DIS.PROP.CD',),
 ('DT.DIS.PRVT.CD',),
 ('DT.INT.BLAT.CD',),
 ('DT.INT.DLXF.CD',),
 ('DT.INT.DPNG.CD',),
 ('DT.INT.MLAT.CD',),
 ('DT.INT.OFFT.CD',),
 ('DT.INT.PBND.CD',),
 ('DT.INT.PCBK.CD',),
 ('DT.INT.PROP.CD',),
 ('DT.INT.PRVT.CD',)]

## 4. The total amount of debt owed by the countries
As previously noted, the financial debt of a specific country reflects its economic condition. However, if we want to examine this on a global scale, how should we proceed?

Let's shift our focus from the debt indicators and calculate the total amount of debt (in USD) owed by various countries. This will provide insight into the overall state of the global economy.

In [ ]:
%%sql
SELECT
ROUND(SUM(debt)/1000000, 2) AS total_debt
FROM international_debt;

 * postgresql:///international_debt
1 rows affected.


[(Decimal('3079734.49'),)]

## 5. Country with the highest debt

That totals over 3 trillion USD, an amount that is difficult to comprehend.

Now that we have determined the precise total of debt owed by various countries, let's identify which country has the highest amount of debt and the specific figure. This debt represents the cumulative total across different categories owed by that country. Understanding this will provide deeper insights into the country's socio-economic conditions. We can also explore which category accounts for the highest debt for that country, but we will set that aside for the moment.

In [ ]:
%%sql

SELECT
country_name , SUM(debt) AS total_debt
FROM international_debt
GROUP BY country_name
ORDER BY total_debt DESC
LIMIT 1;

 * postgresql:///international_debt
1 rows affected.


[('China', Decimal('285793494734.200001568'))]

## 6. Average amount of debt across indicators
It was China.

Now that we have a general overview of the dataset and some summary statistics, we have a clearer understanding of the various debt indicators that countries are associated with. Next, we can delve deeper to determine the average amount of debt owed by a country. This will help us better understand how debt is distributed across different indicators."

In [ ]:
%%sql
SELECT
    indicator_code  AS debt_indicator,
    indicator_name ,
    AVG(debt) AS average_debt
FROM international_debt
GROUP BY debt_indicator, indicator_name
ORDER BY average_debt DESC
LIMIT 10;

 * postgresql:///international_debt
10 rows affected.


[('DT.AMT.DLXF.CD', 'Principal repayments on external debt, long-term (AMT, current US$)', Decimal('5904868401.499193612')),
 ('DT.AMT.DPNG.CD', 'Principal repayments on external debt, private nonguaranteed (PNG) (AMT, current US$)', Decimal('5161194333.812658349')),
 ('DT.DIS.DLXF.CD', 'Disbursements on external debt, long-term (DIS, current US$)', Decimal('2152041216.890243888')),
 ('DT.DIS.OFFT.CD', 'PPG, official creditors (DIS, current US$)', Decimal('1958983452.859836046')),
 ('DT.AMT.PRVT.CD', 'PPG, private creditors (AMT, current US$)', Decimal('1803694101.963265321')),
 ('DT.INT.DLXF.CD', 'Interest payments on external debt, long-term (INT, current US$)', Decimal('1644024067.650806481')),
 ('DT.DIS.BLAT.CD', 'PPG, bilateral (DIS, current US$)', Decimal('1223139290.398230108')),
 ('DT.INT.DPNG.CD', 'Interest payments on external debt, private nonguaranteed (PNG) (INT, current US$)', Decimal('1220410844.421518983')),
 ('DT.AMT.OFFT.CD', 'PPG, official creditors (AMT, current US$)', Decimal('1191187963.083064523')),
 ('DT.AMT.PBND.CD', 'PPG, bonds (AMT, current US$)', Decimal('1082623947.653623188'))]

## 7. The most common debt indicator
China holds the highest debt in the long-term debt (DT.AMT.DLXF.CD) category, as confirmed by The World Bank. It's always a good practice to verify our analyses like this, as it helps ensure that our findings are accurate.

We've seen that long-term debt is the leading category when it comes to average debt amounts. But I wonder if it's also the most common type of debt that countries carry.

In [ ]:
%%sql

SELECT indicator_code, COUNT(indicator_code) AS indicator_count
FROM international_debt
GROUP BY indicator_code
ORDER BY indicator_count DESC, indicator_code DESC
LIMIT 20;


 * postgresql:///international_debt
20 rows affected.


[('DT.INT.OFFT.CD', 124),
 ('DT.INT.MLAT.CD', 124),
 ('DT.INT.DLXF.CD', 124),
 ('DT.AMT.OFFT.CD', 124),
 ('DT.AMT.MLAT.CD', 124),
 ('DT.AMT.DLXF.CD', 124),
 ('DT.DIS.DLXF.CD', 123),
 ('DT.INT.BLAT.CD', 122),
 ('DT.DIS.OFFT.CD', 122),
 ('DT.AMT.BLAT.CD', 122),
 ('DT.DIS.MLAT.CD', 120),
 ('DT.DIS.BLAT.CD', 113),
 ('DT.INT.PRVT.CD', 98),
 ('DT.AMT.PRVT.CD', 98),
 ('DT.INT.PCBK.CD', 84),
 ('DT.AMT.PCBK.CD', 84),
 ('DT.INT.DPNG.CD', 79),
 ('DT.AMT.DPNG.CD', 79),
 ('DT.INT.PBND.CD', 69),
 ('DT.AMT.PBND.CD', 69)]